resources/lib/viervijfzes/content.py

```diff
@@ -375,6 +377,69 @@ def get_categories(self, channel):
 
         return categories
 
+    def get_weather(self, channel):
+        """ Get a weather dictionary.
+        :type channel: str
+        :rtype dict
+        """
+        response = self._get_url(self.SITE_APIS[channel] + '/weather', authentication=True)
+        weather = json.loads(response)
+        return weather
+
+    def get_ad_streams(self, channel, program, path, uuid, video_type, username):
+        """ Get a list of advertisement stream URLs to use for this video.
+        :type channel: str
+        :type path: str
+        :rtype list
+        """
+        ad_streams = []
+        ad_url = 'https://pubads.g.doubleclick.net/gampad/ads'
+        weather = self.get_weather(channel)
+        channel_info = dict(
+            vier=dict(cmsid='2493239', network_id='21797861328'),
+            vijf=dict(cmsid='2493512', network_id='21797861328'),
+            zes=dict(cmsid='2496240', network_id='21797861328')
+        )
+        network_id = channel_info.get(channel).get('network_id')
+        from unicodedata import normalize
+        program = normalize('NFD', program).replace(' ', '-')
+        program = re.sub(r'[^A-Za-z0-9-]+', '', program).lower()
+        from hashlib import sha1
+        ppid = sha1(username).hexdigest()
+        if program:
+            iu_id = '/{}/{}/{}/{}'.format(network_id, channel, 'programmas', program)
+        else:
+            iu_id = '/{}/{}/'.format(network_id, channel)
+        params = dict(ad_rule=1,
+                      cmsid=channel_info.get(channel).get('cmsid'),
+                      correlator=int(round(time.time() * 1000)),
+                      sbs_weather_cond=weather.get('summary'),
+                      sbs_weather_temp=weather.get('temperature'),
+                      description_url=path,
+                      env='vp',
+                      gdfp_req=1,
+                      impl='s',
+                      iu=iu_id,
+                      output='vast',
+                      pp='SBSNoDash',
+                      ppid=ppid,
+                      sz='640x360',
```

Comment:
Does this define the resolution of the ads? Is a higher resolution available?

Reply:
I just copied the parameters that were hardcoded in javascript on vier.be. You can find the documentation here: https://support.google.com/admanager/answer/7320899
1080p is available in the HLS manifest and is selected by default by InputStream Adaptive I think.